dht_net.go

```diff
@@ -178,6 +179,7 @@ func (ms *messageSender) SendMessage(ctx context.Context, pmes *pb.Message) erro
 }
 
 func (ms *messageSender) writeMessage(pmes *pb.Message) error {
+	ms.s.SetWriteDeadline(time.Now().Add(dhtMessageTimeout))
```

Comment:
Is the writeDeadline scoped per WriteMsg() call? Otherwise I think we might have to clear the deadline when we're finished with this writeMessage() call.